lib/format/duration/formatters/default.ex

```diff
@@ -80,11 +82,13 @@ defmodule Timex.Format.Duration.Formatters.Default do
   defp format_component({:minutes, m}, str), do: str <> "#{m}M"
   defp format_component({:seconds, s}, str), do: str <> "#{s}S"
 
-  defp deconstruct(%Duration{microseconds: micro} = duration),
-    do: deconstruct({Duration.to_seconds(duration, truncate: true), rem(micro, 1_000_000)}, [])
+  defp deconstruct(duration) do
+    micros = Duration.to_microseconds(duration) |> abs
```

Comment:
I switched to microseconds here because there was a bug with formatting of durations that had positive seconds and negative micros or vice versa. For instance, in previous implementation `Duration.from_erl({0, -2, 1_100_000})` was formatted as `PT0.1S` rather than `PT0.9S`. I added a [corresponding test](https://github.com/bitwalker/timex/pull/265/files#diff-0ff0a1e5c7cc6ecb09879df8a6b3fdadR16) for this case